### PR TITLE
Release 3.0.0

### PR DESCRIPTION
**Pull request by zhusee2**
# Changes in this release 
- Refactor `closable()` mixin to use React's own events mechanism #202
- Enhance `<TextInputRow>` for custom rendering and textarea autosizing #203
- Remove storybook mangle #204
- Refactor `<TextInput>` to match latest design #205
- Fix yarn start not work due to adding minimizer in dev env #206
- Add `<Avatar>` to display an image along with the text #208
- Enhance `<ImageEditor>` for get canvas and controllable scale #209
- Allow passing children to <TextInputRow> #trivial #211 _(since beta 1)_

# Prereleases
- `3.0.0-beta.0`
- `3.0.0-beta.1`

# Changelog
### Breaking
- [Core] Add `verticalOrder` prop to `<Text>` so you can swap the position of `basic` and `aside`. Also applied to `rowComp()` mixin.
- [Core] Rewrite `<TextInput>` to match latest design, offering single-line `<input>`, multi-line `<textarea>` and supports custom rendering via render prop.
- [Form] `<TextInputRow>` now renders the new `<TextInput>` and forwards almost every prop to it, **without** a ref to its inner input.
- [Core] Add `<Avatar>` to display an image.

### Changed
- [ImageEditor] Add new instance method `getImageCanvas()` to get current image canvas element.
- [ImageEditor] Add new props `scale` & `onScaleChange` to make scale value of editor can be controlled.
- [Core] Refactored `closable()` mixin to detect inside/outside clicks via React SyntheticEvent mechanism instead of listening native events from DOM.
- [Storybook] Fix mangled component name in storybook build. (#203)
- [Storybook] Update examples for core `<TextInput>` and form `<TextInputRow>`. (#203)
- [Core] Change `rowComp()` to allow the appearance of `<Avatar>` alongside the text. (#208)
- [Core] Change `<Checkbox>` to display `<Avatar>`. (#208)
- [Form] Change `<SelectRow>` and `<Checkbox>` to display `<Avatar>`. (#208)
- [Storybook] Add examples for `<Avatar>` and the list components with `<Avatar>`s. (#208)

### CODE REVIEW

File: packages/form/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@ichef/gypcrete-form",
-  "version": "2.1.0",
+  "version": "3.0.0-beta.1",
```
Comment (garfieldduck) [inline packages/form/package.json]: Should it be `3.0.0`?

File: packages/form/package.json [hunk 2]
```diff
@@ -34,10 +34,11 @@
     "react": "^16.6.0"
   },
   "dependencies": {
-    "@ichef/gypcrete": "^2.1.0",
+    "@ichef/gypcrete": "^3.0.0-beta.1",
```
Comment (garfieldduck) [inline packages/form/package.json]:
```suggestion
    "@ichef/gypcrete": "^3.0.0",
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+### Breaking
```
Comment (tz5514) [inline CHANGELOG.md]: @zhusee2 Is here should be `[3.0.0]`?

**Comment by zhusee2** [comment]
Ah thanks for reminding, I'll bump the version before merge.

**Pull request merged**